### PR TITLE
Add pstreams library

### PR DESCRIPTION
**Pull request by arthurl**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -591,6 +591,12 @@
       fingerprint = "3D2B B230 F9FA F0C5 1832  46DD 4FDC 96F1 61E7 BA8A";
     }];
   };
+  arthur = {
```
Comment (multun) [inline maintainers/maintainer-list.nix]: Can you move it to a separate commit ?

File: pkgs/development/libraries/pstreams/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  dontBuild = true;
+  doCheck = true;
+
+  meta = with stdenv.lib; {
+    description = "POSIX Process Control in C++";
+    longDescription = ''
+      PStreams allows you to run another program from your C++ application and
+      to transfer data between the two programs similar to shell pipelines.
+
+      In the simplest case, a PStreams class is like a C++ wrapper for the
```
Comment (multun) [inline pkgs/development/libraries/pstreams/default.nix]: This description is way too long.
long descriptions usually look like this https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/gcc/9/default.nix#L364

**Review by multun — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/pstreams/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{ stdenv
+, fetchgit
+}:
+
+stdenv.mkDerivation rec {
+  pname = "PStreams";
+  version = "d21afef311be496e6495a5e2103eec7a53ecb817";
```
Comment (multun) [inline pkgs/development/libraries/pstreams/default.nix]: This version string is way too long. How about `d21afef` or even `1.0.1-d21afef` ?

**Review by multun — CHANGES_REQUESTED** [review]

**Comment by multun** [comment]
Would you mind fixing [the commit message format](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes)?

File: pkgs/development/libraries/pstreams/default.nix (hunk 3 above)
Comment (c0bw3b) [reply pkgs/development/libraries/pstreams/default.nix]: Even better to stick to the stable release `1.0.1`.
The only commit that comes after doesn't seem vital and may even break programs depending on `pstreams`

**Comment by c0bw3b** [comment]
Thanks for your first contribution to nixpkgs and welcome around here @arthurl

File: pkgs/development/libraries/pstreams/default.nix (hunk 2 above)
Comment (c0bw3b) [reply pkgs/development/libraries/pstreams/default.nix]: We don't really set a maximum length [in the manual](https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes). We have other packages with truly long `longDescription` (cf factorio or hedgewars) and it still renders properly on nixos.org package listing.

But here we can probably lose the last paragraph and still retain the essential bits of information.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (arthurl) [reply maintainers/maintainer-list.nix]: done.

File: pkgs/development/libraries/pstreams/default.nix (hunk 2 above)
Comment (arthurl) [reply pkgs/development/libraries/pstreams/default.nix]: Agreed. No point providing internal details here.

File: pkgs/development/libraries/pstreams/default.nix (hunk 3 above)
Comment (arthurl) [reply pkgs/development/libraries/pstreams/default.nix]: Good point. The other commits apart from the last one that you've already mentioned are all testing related. Not much point moving beyond 1.0.1.

**Comment by arthurl** [comment]
> Would you mind fixing [the commit message format](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes)?

Fixed.

I've also added a fix to get it working on Darwin.

**Comment by arthurl** [comment]
> Thanks for your first contribution to nixpkgs and welcome around here @arthurl

Thanks!

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build pstreams

**Review by c0bw3b — APPROVED** [review]
